onset_maproom/calc.py

```diff
@@ -268,6 +268,23 @@ def seasonal_sum(
     summed_seasons = xr.merge([seasonal_data, seasons_ends])
     return summed_seasons
 
+def probExceed(onsetMD, earlyStart):
+    onsetMD_df = onsetMD
+    onsetDatetime = pd.to_datetime(onsetMD_df.onset)
+    onsetDiff = onsetDatetime - earlyStart
+    onsetDiff_df = onsetDiff.to_frame()
+    counts = onsetDiff_df['onset'].value_counts()
+    countsDF = counts.to_frame().sort_index()
+    cumsum = countsDF.cumsum()
+    onset = []
+    for i in onsetDiff_df.T:
+        float = onsetDiff_df.onset[i].total_seconds() / 86400
+        onset.append(float)
```

Comment:
Can lines 279-282 be rewritten in one line?
```
onset = onsetDiff_df.onset.dt.total_seconds() / (24 * 60 * 60)
```
When there's a choice between using a python loop and using matrix algebra, the matrix algebra approach is generally preferable . Besides being more succinct, it's much faster. (That said, if you find it easier to think in terms of loops, go ahead and write it that way first. Once you get it right, you can rewrite it in a more efficient way and then verify that the result is the same.)